zulipterminal/ui_tools/boxes.py

```diff
@@ -82,19 +83,35 @@ def stream_box_view(self, button: Any=None, caption: str='',
 
     def keypress(self, size: Tuple[int, int], key: str) -> str:
         if is_command_key('SEND_MESSAGE', key):
-            if not self.to_write_box:
-                success = self.model.send_stream_message(
-                    stream=self.stream_write_box.edit_text,
-                    topic=self.title_write_box.edit_text,
-                    content=self.msg_write_box.edit_text
-                )
+            if self.msg_edit_id:
+                if not self.to_write_box:
+                    success = self.model.update_stream_message(
```

Comment:
While this definitely works, I thought it a little odd to check whether the 'topics' box has been set (or None) to determine whether it is a stream / PM. Maybe we could check the message type like always? Just a suggestion (v2?)

Reply:
We will have to provide writebox with the whole msg, so not sure about this.

Reply:
I think this is reasonable to keep for now, but perhaps we can look into this later - this isn't as obvious as it could be, certainly.